### PR TITLE
Implement mapping-binding-pattern, capture-binding-pattern in incremental-parser

### LINKED ISSUE #23948: Implement mapping-binding-pattern, capture-binding-pattern in incremental-parser
**Description:**
$title

**Suggested Labels (optional):**
component/parser

**Suggested Assignees (optional):**
@dulvinw

### PR DESCRIPTION
**Pull request by dulvinw**
Fixes #23948

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -1845,6 +1851,18 @@ public BLangLetVariable createLetVariable(LetVariableDeclarationNode letVarDecl)
         return letVar;
     }
 
+    @Override
+    public BLangNode transform(CaptureBindingPatternNode captureBindingPatternNode) {
+        Node parent = captureBindingPatternNode.parent();
+        if (parent instanceof FieldBindingPatternVarnameNode || parent instanceof FieldBindingPatternFullNode
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1857]: can we use the kind please.

Comment (dulvinw) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1857]: Kind is the same for both of these :( So went for instanceof. Shall I change the kind as well?

Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1857]: @SupunS shouldn't we have different kinds?

Comment (dulvinw) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Fixed

Comment (rasika) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Seems like only `FieldBindingPatternVarnameNode` and `FieldBindingPatternFullNode` has `FIELD_BINDING_PATTERN` kind. So we can go with kind is guess.

**Pull request merged**